src/iothread.c

```diff
@@ -28,6 +50,9 @@ void enqueuePendingClientsToMainThread(client *c, int unbind) {
     if (unbind) connUnbindEventLoop(c->conn);
     /* Just skip if it already is transferred. */
     if (c->io_thread_client_list_node) {
+        /* If there are several clients to process, let the main thread handle them ASAP. */
+        sendPendingClientsToMainThreadIfNeeded(&IOThreads[c->tid], 1);
```

Comment:
why not put it after `listAddNodeTail(IOThreads[c->tid].pending_clients_to_main_thread, c);`

Reply:
Thank you, this is important, i added a comment about the reason

>         /* If there are several clients to process, let the main thread handle them ASAP.
>          * Since the client being added to the queue may still need to be processed by
>          * the IO thread, we must call this before adding it to the queue to avoid
>          * races with the main thread. */